api/src/test/java/org/openmrs/validator/VisitValidatorTest.java

```diff
@@ -98,6 +99,77 @@ public void validate_shouldRejectAVisitIfItHasFewerThanMinOccursOfAnAttribute()
 	
 	/**
 	 * @see VisitValidator#validate(Object,Errors)
+	 * @verifies reject a visit if it has more than max occurs of an attribute when expected exactly one
+	 */
+	@Test
+	public void validate_shouldRejectAVisitIfItHasMoreThanMaxOccursOfAnAttributeWhenExpectedExactlyOne() throws Exception {
+		executeDataSet("org/openmrs/validator/include/VisitValidatorTest-additionalAttributeType1.xml");
+		
+		Visit visit = makeVisit();
+		visit.addAttribute(makeAttribute("one"));
+		visit.addAttribute(makeAttribute("two"));
+		visit.addAttribute(makeAttribute("three"));
+		visit.addAttribute(makeAttribute("four"));
+		try {
+			ValidateUtil.validate(visit);
+		}
+		catch (ValidationException e) {
+			Errors errors = e.getErrors();
+			assertEquals(1, errors.getErrorCount());
+			assertEquals("attribute.error.maxOccurs", errors.getFieldError("activeAttributes").getCode());
+		}
+	}
+	
+	/**
+	 * @see VisitValidator#validate(Object,Errors)
+	 * @verifies reject a visit if it has less than min occurs of an attribute when expected exactly one
+	 */
+	@Test
+	public void validate_shouldRejectAVisitIfItHasLessThanMinOccursOfAnAttributeWhenExpectedExactlyOne() throws Exception {
+		executeDataSet("org/openmrs/validator/include/VisitValidatorTest-additionalAttributeType1.xml");
+		
+		Visit visit = makeVisit();
+		try {
+			ValidateUtil.validate(visit);
+		}
+		catch (ValidationException e) {
+			Errors errors = e.getErrors();
+			assertEquals(1, errors.getErrorCount());
```

Comment:
good point, you are perfectly right! I'll add a fail(), in case the Validation succeeds